Concurrency for GlassFish 8

dmatej opened the pull request:
Draft PR depending on work of @aubi , see https://github.com/eclipse-ee4j/glassfish-concurro/pull/85

Requires also new Concurrency TCK 3.1.1:
https://github.com/jakartaee/concurrency/issues/511

dmatej committed:
Added descriptors for JEE 11

Signed-off-by: David Matějček <david.matejcek@omnifish.ee>

dmatej committed:
Updated dependency on concurrency TCK

Signed-off-by: David Matějček <david.matejcek@omnifish.ee>

dmatej committed:
Added qualifiers annotation attribute processing

Signed-off-by: David Matějček <david.matejcek@omnifish.ee>

dmatej committed:
Dependency on Concurro snapshot version

Signed-off-by: David Matějček <david.matejcek@omnifish.ee>

dmatej committed:
Added the attribute 'virtual', new in Concurrency 3.1

Signed-off-by: David Matějček <david.matejcek@omnifish.ee>

dmatej committed:
Fixed NPE

Signed-off-by: David Matějček <david.matejcek@omnifish.ee>

aubi committed:
temporarily disable concurro dependency due to osgi problems

dmatej committed:
More robust work with qualifiers in descriptors and annotations

Signed-off-by: David Matějček <david.matejcek@omnifish.ee>

dmatej committed:
Fixed classloader issues - qualifiers must be processed as strings in DOL

- Classifiers may be distributed in a library which is not accessible to the
  deployment classloader yet.

Signed-off-by: David Matějček <david.matejcek@omnifish.ee>

aubi committed:
never override qualifiers by annotation in concurrency definitions

dmatej committed:
Revert "never override qualifiers by annotation in concurrency definitions"

This reverts commit 2615369dda56c4e8d01b6963a8c2bd55d2f14b7d.

dmatej committed:
The ContextServiceDefinitionConverter improved

- Now follows the same pattern as other concurrency converters

Signed-off-by: David Matějček <david.matejcek@omnifish.ee>

dmatej committed:
ContextServiceDefinitionDescriptor adapted to ContextualResourceDefinition

Signed-off-by: David Matějček <david.matejcek@omnifish.ee>

dmatej committed:
Added more detailed toStrings for concurrency data objects

Signed-off-by: David Matějček <david.matejcek@omnifish.ee>

dmatej committed:
Added log to ComponentEnvManagerImpl

dmatej committed:
Correct merging qualifiers respecting a special empty element

- Depends on elementsAllowingEmptyValue() specified by a root node, but now
  I am not sure which is the right one. That part will be committed tomorrow.

Signed-off-by: David Matějček <david.matejcek@omnifish.ee>

dmatej committed:
Removing an empty String which may occur in XML as a special value

Signed-off-by: David Matějček <david.matejcek@omnifish.ee>

dmatej committed:
ApplicationNode can contain empty qualifier element

Signed-off-by: David Matějček <david.matejcek@omnifish.ee>

aubi commented:
This branch waits either for (preferably)  a release of Concurro and update of the dependency or staging Concurro SNAPSHOT.

dmatej committed:
integrate concurro and concurrency 3.1

# Conflicts:
#	appserver/common/glassfish-naming/src/main/java/com/sun/enterprise/naming/impl/GlassfishNamingManagerImpl.java
#	appserver/deployment/dol/src/main/java/com/sun/enterprise/deployment/util/DOLUtils.java

dmatej committed:
Cleanups around deployments and JNDI, using module name in ComponentInvocation

Signed-off-by: David Matějček <david.matejcek@omnifish.ee>

# Conflicts:
#	appserver/web/weld-integration/src/main/java/org/glassfish/weld/WeldDeployer.java

dmatej committed:
Fixed lookup for java:app/ contexts

- This fixed the last concurrency TCK test

Signed-off-by: David Matějček <david.matejcek@omnifish.ee>

arjantijms committed:
Update pom.xml

Use Concurro 3.1.0-M4 instead of SNAPSHOT

dmatej commented:
It waits also for #24962 as I need to rebase this branch then and remove duplicities. I already done that locally, but it is safer to merge to master first.

dmatej committed:
Referring to custom built Concurrency TCK 3.1.1-SNAPSHOT

- With 3.1.0 several tests fail
- Required PRs were already merged to the TCK main, but we have to wait for
  the release.

Signed-off-by: David Matějček <david.matejcek@omnifish.ee>
